sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/DocumentsImpl.java

```diff
@@ -113,7 +112,7 @@
                 @QueryParam("$select") String select,
                 @QueryParam("$skip") Integer skip,
                 @QueryParam("$top") Integer top,
-                @QueryParam("captions") Captions captions,
```

Comment:
Captions seems to have been renamed to QueryCaptions, why have this param as a string then?

Reply:
`Captions`, and `Answers`, are a hard query parameter to deal with when it comes to code generation. It is declared as an expandable string enum in Swagger with a few pre-defined values, but if you checkout the REST API specs documentation it uses its own special concatenation to send additional data to the service. For example, at a top level your captioning can be `NONE` or `EXTRACTIVE`, effectively have captions or not, but then based on the docs you can append `|highlight-{boolean}` to indicate if the extracted captions should include highlighting.

That doesn't generate well, nor do I believe it should as this is a very specialized scenario. So, from this we use a String here instead of `Captions`/`QueryCaption`.

Reply:
So, should we not work with the team and change the swagger and make the API take in a string? We need to have the type in the swagger but may be not as the query param type?

Reply:
Unfortunately, this would be a breaking change for Swagger